File: branches/Medusa/Source/Fantasy.Jobs/Resources/TemplateRuntimeScheduleResourceProvider.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Fantasy.Jobs.Management;

namespace Fantasy.Jobs.Resources
{
    public class TemplateRuntimeScheduleResourceProvider : RuntimeScheduleResourceProviderBase
    {
        public TemplateRuntimeScheduleResourceProvider()
        {

        }
        protected override bool InternalCanHandle(string name)
        {
            return string.Equals("RunJob", name, StringComparison.OrdinalIgnoreCase);
        }

        public override void Initialize()
        {
            base.Initialize();
            JobManagerSettings.Default.PropertyChanged += new System.ComponentModel.PropertyChangedEventHandler(SettingsPropertyChanged);
        }


        void SettingsPropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
        {
            if (e.PropertyName == "TemplateRuntimeSchedule")
            {
                this.Reschedule();
            }
        }

        protected override RuntimeScheduleSetting GetSetting(string id)
        {
            RuntimeScheduleSetting rs = JobManagerSettings.Default.TemplateRuntimeSchedule.Templates.Find(s => string.Equals(s.Name, id, StringComparison.OrdinalIgnoreCase));
            if (rs == null)
            {
                rs = JobManagerSettings.Default.TemplateRuntimeSchedule;
            }

            return rs;
        }

        protected override string GetResourceId(ResourceParameter parameter)
        {
            return parameter.Values["template"];
        }
    }
}

File: Fantasy.Jobs/src/fantasy/jobs/resources/TemplateRuntimeScheduleResourceProvider.java
package fantasy.jobs.resources;

import Fantasy.Jobs.Management.*;

public class TemplateRuntimeScheduleResourceProvider extends RuntimeScheduleResourceProviderBase
{
	public TemplateRuntimeScheduleResourceProvider()
	{

	}
	@Override
	protected boolean InternalCanHandle(String name)
	{
		return String.equals("RunJob", name, StringComparison.OrdinalIgnoreCase);
	}

	@Override
	public void Initialize()
	{
		super.Initialize();
//C# TO JAVA CONVERTER TODO TASK: Java has no equivalent to C#-style event wireups:
		JobManagerSettings.Default.PropertyChanged += new System.ComponentModel.PropertyChangedEventHandler(SettingsPropertyChanged);
	}


	private void SettingsPropertyChanged(Object sender, System.ComponentModel.PropertyChangedEventArgs e)
	{
		if (e.PropertyName.equals("TemplateRuntimeSchedule"))
		{
			this.Reschedule();
		}
	}

	@Override
	protected RuntimeScheduleSetting GetSetting(String id)
	{
//C# TO JAVA CONVERTER TODO TASK: Lambda expressions and anonymous methods are not converted by C# to Java Converter:
		RuntimeScheduleSetting rs = JobManagerSettings.getDefault().getTemplateRuntimeSchedule().getTemplates().Find(s => String.equals(s.getName(), id, StringComparison.OrdinalIgnoreCase));
		if (rs == null)
		{
			rs = JobManagerSettings.getDefault().getTemplateRuntimeSchedule();
		}

		return rs;
	}

	@Override
	protected String GetResourceId(ResourceParameter parameter)
	{
		return parameter.getValues().get("template");
	}
}